bin/syclcc-clang

```diff
@@ -558,7 +558,11 @@ class pure_cpu_compiler:
     hipsycl_library_path = os.path.join(config.hipsycl_installation_path,"lib/")
 
     self._linker_args = []
-    self._compiler_args = ["-fopenmp"]
+    self._compiler_args = []
+    if sys.platform == "darwin":
+        # y'all still need libomp installed
+        self._compiler_args += ["-Xclang"]
+    self._compiler_args += ["-fopenmp"]
```

Comment:
I have very little experience on Mac - I know that the default compiler there is clang, but I wonder if this is really the only choice for an OpenMP compiler? The hipSYCL CPU backend can work with any C++ OpenMP compiler. Would Mac users be interested in e.g. using gcc on Mac? If so, we would need to change this so that it doesn't break other compilers.

Reply:
I am also a Linux users and just support a lot of developers in our team running on macs. Generally, most popular open source projects just focus on supporting & testing the native system compiler on macOS (and Windows) - AppleClang is the primary compiler there.

Anything else (e.g. GCC stacks on macOS) are more challenging and get quickly into nifty support problems, simply because most upstream projects don't test against it. I don't see how the current logic works with other compilers than clang right now and I think this would need on all systems more abstraction as yet implemented, independent of macOS.

Reply:
Okay, in that case I think it's fine for the time being. On Linux, it just works with multiple compilers by virtue of `-fopenmp` being the only thing necessary to get it working with gcc, clang, icc etc. Thus switching the compiler with `HIPSYCL_CPU_CXX` just works. I think the "correct" way to generalize this to handle such cases would be to to give the user the ability to customize the compilation command as we have already done for the linking steps on CUDA and ROCm with the `CUDA_LINK_LINE` and `ROCM_LINK_LINE` settings.